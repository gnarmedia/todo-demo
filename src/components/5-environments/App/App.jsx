// import dependencies
import React, { Fragment } from "react";

// import components
import TaskList from "../../3-organisms/TaskList/TaskList.jsx";

// mock up a list of tasks, be sure the ids are unique
const localTasks = localStorage.getItem("tasks"),
    tasks = localTasks
        ? JSON.parse(localStorage.getItem("tasks"))
        : [
              { id: "ryBRDzBdM", title: "Task 1", isComplete: false },
              { id: "HygBRPGBuG", title: "Task 2", isComplete: true },
              { id: "r1ZSCDMruM", title: "Task 3", isComplete: false }
          ];

/**
 * @name App
 * @desc Renders the markup for our app
 * @returns {function} App component
 */
const App = () => (
    <Fragment>
        <h1>Task List demo</h1>
        <TaskList tasks={tasks} filter="ALL" />
    </Fragment>
);

// export our component for importing into other modules
export default App;
