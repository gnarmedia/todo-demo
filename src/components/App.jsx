// import dependencies
import React, { Fragment } from "react";

// import components
import TaskList from "./TaskList.jsx";

// mock up a list of tasks, be sure the ids are unique
const localTasks = localStorage.getItem("tasks"),
    tasks = (localTasks ? 
        JSON.parse(localStorage.getItem("tasks")) :
        [
            { id: "ryBRDzBdM", title: "Task 1", isComplete: false },
            { id: "HygBRPGBuG", title: "Task 2", isComplete: true },
            { id: "r1ZSCDMruM", title: "Task 3", isComplete: false }
        ]
    );

/**
 * @name: App
 * @desc: This is a function that renders the markup for our app, for brevity it
 *        is all done in a single component.
 * @returns: {function}
 */
const App = () => (
    <Fragment>
        <h1>Task List demo</h1>
        <TaskList tasks={tasks} filter="ALL" />
    </Fragment>
);

// export our component for importing into other modules
export default App;
