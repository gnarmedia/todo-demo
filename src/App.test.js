import React from 'react';
import renderer from 'react-test-renderer';

// import component
import App from './App.jsx';

describe('App', () => {
    it('should render the App', () => {
        const tree = renderer.create(<App />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
