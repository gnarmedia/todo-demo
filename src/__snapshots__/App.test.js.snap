// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App 1`] = `
Array [
  <h1>
    Task List demo
</h1>,
  <ul>
    <li>
        <input
            defaultChecked={false}
            type="checkbox"
        />
        Task 1
         [
        incomplete
        ]
    </li>
    <li>
        <input
            defaultChecked={true}
            type="checkbox"
        />
        Task 2
         [
        complete
        ]
    </li>
    <li>
        <input
            defaultChecked={false}
            type="checkbox"
        />
        Task 3
         [
        incomplete
        ]
    </li>
</ul>,
]
`;
